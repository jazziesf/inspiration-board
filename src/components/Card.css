.card {
  background-color: #F8B195;

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  align-items: center;
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  font-size: 2em;
  color: white;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete {
  align-self: start;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  font-size: 2em;
}

.card__delete:hover {
  align-self: start;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  font-size: 3em;
  cursor: pointer;
}
