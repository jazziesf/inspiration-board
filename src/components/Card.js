import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getUnicode } from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  renderEmoji(emoji) {
    return <p className="card__content-emoji">{getUnicode(emoji)}</p>
  }

  renderText(text) {
    return <p className="card__content-text">{text}</p>
  }


  render() {

    const icon = this.props.emoji ? this.renderEmoji(this.props.emoji) : null
    const text = this.props.text ? this.renderText(this.props.text) : null


    return (
      <div className="card">
        <div className="card__content">
          {text}
        
          {icon}

        </div>
      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string,
};

export default Card;
