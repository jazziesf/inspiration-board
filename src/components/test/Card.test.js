import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('inspiration card', () => {

  it('will match the inspirationCard Snapshot', () => {
      const wrapper = shallow( <Card
        text="hi"
        emoji=""
        id={1}
        onDeleteCallback={() => {}}
        />);
    expect(wrapper).toMatchSnapshot();
  });


})
