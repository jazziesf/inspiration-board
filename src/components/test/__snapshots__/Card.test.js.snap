// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inspiration card will match the inspirationCard Snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <p
      className="card__content-text"
    >
      hi
    </p>
  </div>
  <div
    className="card__delete"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
